File: ruby/testcode/massassign_046.rb
require_relative 'shared'

User = Struct.new(:name, :email, :role, :bio, :admin, keyword_init: true)
class << User; def create(h); new(**h.transform_keys(&:to_sym).slice(:name,:email,:role,:bio,:admin)); end; end

# vuln-code-snippet start ruby_massassign_no_direct_params
def handler(req)
  name  = req.post('name')
  email = req.post('email') # vuln-code-snippet safe-line ruby_massassign_no_direct_params
  user  = User.new(name: name, email: email)
  BenchmarkResponse.json({ ok: true })
end
# vuln-code-snippet end ruby_massassign_no_direct_params

File: php/testcode/massassign_046.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_massassign_event_sourcing_immutable
function massassign046(BenchmarkRequest $req): BenchmarkResponse {
    $cmd = new stdClass();
    $cmd->name = $_POST['name'] ?? ''; // vuln-code-snippet safe-line php_massassign_event_sourcing_immutable
    $cmd->email = $_POST['email'] ?? '';
    return BenchmarkResponse::ok('command dispatched');
}
// vuln-code-snippet end php_massassign_event_sourcing_immutable
